staging/src/k8s.io/component-base/featuregate/feature_gate.go

```diff
@@ -309,7 +318,27 @@ func (f *featureGate) Validate() []error {
 		return []error{fmt.Errorf("cannot cast enabledRaw to map[string]bool")}
 	}
 	enabled := map[Feature]bool{}
-	return f.unsafeSetFromMap(enabled, m, f.EmulationVersion())
+	errs := f.unsafeSetFromMap(enabled, m, f.EmulationVersion())
+
+	dependencies := *f.dependencies.Load()
+	known := f.known.Load().(map[Feature]VersionedSpecs)
+	for feature, deps := range dependencies {
+		if !featureEnabled(feature, enabled, known, f.EmulationVersion()) {
+			continue
+		}
+
+		var disabledDeps []Feature
+		for _, dep := range deps {
+			if !featureEnabled(dep, enabled, known, f.EmulationVersion()) {
+				disabledDeps = append(disabledDeps, dep)
+			}
+		}
+		if len(disabledDeps) > 0 {
+			errs = append(errs, fmt.Errorf("cannot enable %s since it depends on disabled features: %v", feature, disabledDeps))
```

Comment:
nit: maybe tweak the error message to suggest the resolution instead of telling them they cannot enable X

```suggestion
			errs = append(errs, fmt.Errorf("%s is enabled, but depends on features which are disabled: %v", feature, disabledDeps))
```